jib-cli/build.gradle

```diff
@@ -0,0 +1,36 @@
+plugins {
+  id "application"
+}
+
+ext {
+    cliMainClass = "com.google.cloud.tools.jib.cli.JibCli"
+}
+
+// use `run` to build and run the app
+// use `installDist` or `distZip` to create an installable application
+application {
+  applicationName = "jib"
+  mainClassName = cliMainClass
+}
+
+dependencies {
+  implementation project(':jib-build-plan')
```

Comment:
Can you see if you can remove this (as we are copying `jib-build-plan` source into `jib-core`)? `jib-plugins-common` does not add this, for example. @loosebazooka may have a better idea.

Reply:
Yeah we can remove this I think? But I'm not sure how this build works -- does it just package up the application without depending on anything being in the local maven repo?

Reply:
I've tried removing this and the build breaks with unresolved symbols 🤷‍♂

Reply:
oh yeah, because it's not re-exported. so I think even in jib-core we need to make it an `api` dependency rather than `implementation`?

Reply:
So what should be done in this PR?

Reply:
I think it's correct for this PR.